### DISCUSSION TITLE
New Pine64 Pinecil v2 Firmware

### DISCUSSION THREAD

**Discussion by Helblind** [Ralim/IronOS#1984] [General]
Hey all!

I just picked up a new Pine64 Pinecil v2 from the Amazon US store. The manual directed me here to potentially upgrade the firmware. Is this something I should be doing with a brand new device? Is there a way to check the current firmware? If I should update the firmware are there instruction on how? I am familiar with Windows and not with Linux. Thanks!

**Comment by Ralim** [comment]
>Is this something I should be doing with a brand new device?
Its recommended as there can be improvements in newer versions; so its worth checking and to make people aware.

> Is there a way to check the current firmware? 
Press and hold on the rear button of the device (near usb ports) and it brings up diagnostics menu with information

>If I should update the firmware are there instruction on how?

For the V2, I would suggest [PineFlash](https://github.com/Spagett1/PineFlash) as the most user friendly option

**Comment by Helblind** [comment]
Thanks Ralim!

So, it looks like it shipped with v2.20.14DAF70 13-12-22. I'll download PineFlash from your link and update it.

Cheers!

**Comment by Helblind** [comment]
Which version should I go with: v2.22, v2.22-rc3, v2.22-rc2?

**Reply by StevenSeifried** [reply]
v2.22. RC stands for Release Candidate

**Reply by Helblind** [reply]
Great, thanks Steven!
